pyaldata/utils.py

```diff
@@ -625,6 +629,20 @@ def _slice_in_trial(trial, sl, warn=False):
 
     is_inside = True
 
+    if start_point_name is not None:
+        if not np.isfinite(trial[start_point_name]):
+            is_inside = False
+            if warn:
+                warnings.warn(f"Invalid time index on trial with ID {trial.trial_id}. Start index is not finite.")
+    if end_point_name is not None:
+        if not np.isfinite(trial[end_point_name]):
+            is_inside = False
+            if warn:
+                warnings.warn(f"Invalid time index on trial with ID {trial.trial_id}. End index is not finite.")
+    if is_inside is False:
+        return is_inside  # otherwise next line might throw an error
```

Comment:
So this makes sure that the errors I mentioned above are not thrown, right?
I'm not sure this is the best place to handle this. Also, this doesn't help if only `epoch_fun` is passed.

Reply:
yes, exactly.
This function is `_slice_in_trial` so I thought it's better to take care of it here. Where else do you have in mind? Because the rest of this function is also doing checks basically.
Re `epoch_fun` I have not idea as I never used it :grin: